### PR TITLE
[iOS] Implement IsEnabled property on Editor

### LINKED ISSUE #6700: [iOS] Editor.IsEnabled doesn't do anything
The Editor on iOS inherits from UITextView which doesn't inherit from UIControl

so the following code doesn't do anything
https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/iOS/ViewExtensions.cs#L22-L23

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Implement IsEnabled property on iOS Editor.

### Issues Fixed

Fixes #6700

### CODE REVIEW

File: src/Core/src/Platform/iOS/TextViewExtensions.cs [hunk 1]
```diff
@@ -65,6 +65,11 @@ public static void UpdateIsReadOnly(this UITextView textView, IEditor editor)
 			textView.UserInteractionEnabled = !(editor.IsReadOnly || editor.InputTransparent);
 		}
 
+		public static void UpdateIsEnabled(this UITextView textView, IEditor editor)
+		{
+			textView.Editable = editor.IsEnabled;
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/iOS/TextViewExtensions.cs:70]: Should we also disable the selection (`Selectable` property)?

**Comment by mattleibow** [comment]
Why is this an iOS specific? I am wondering if we should rather update the extension method to check this type to not only remove the API break but also because it will be surprising. What if the extension is being called from the base UIView?

**Review by SotoiGhost — APPROVED** [review]
LGTM! 👍🏼

File: src/Core/src/Platform/iOS/TextViewExtensions.cs (hunk 1 above)
Comment (SotoiGhost) [reply src/Core/src/Platform/iOS/TextViewExtensions.cs:70]: at least on iOS, the selection and edition are handled separately. Not sure on other platforms

**Comment by SotoiGhost** [comment]
> What if the extension is being called from the base UIView?

Actually, I have the same doubt, is the MAUI control `ContentView.Enabled` property working on iOS?

**Comment by jsuarezruiz** [comment]
> Why is this an iOS specific? I am wondering if we should rather update the extension method to check this type to not only remove the API break but also because it will be surprising. What if the extension is being called from the base UIView?

On all platforms we manage the IsEnabled property at the ViewHandler level in the MapIsEnabled method, then on each platform managed by the UpdateIsEnabled method of ViewExtensions.

However, in this specific case in Xamarin.Forms it was managed in the Renderer with some workaround https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.Android/Renderers/EditorRenderer.cs#L59
In .NET MAUI we manage it at the Handler level to modify the `Editable `and `Selectable `properties of the UITextView. The extension method expects a control of type UITextView not UIView.

**Review by rmarinho — APPROVED** [review]

**Pull request merged**